### PR TITLE
Overlapping events

### PR DESCRIPTION
**Pull request by JosefForkman**

### CODE REVIEW

File: src/lib/utils.ts [hunk 1]
```diff
@@ -62,19 +76,42 @@ const startOfEvent = (event: EventType) => event.deadlines[0].timestamp
...
+    )
+    const EventAWidth =
+        elementWidthOfDeadlineTitle(eventA.deadlines[0].name) >
+        elementWidthOfTitle(eventA.title)
+            ? elementWidthOfDeadlineTitle(eventA.deadlines[0].name)
+            : elementWidthOfTitle(eventA.title)
+    const startOfEventA = xPosEventA - EventAWidth
+    const endOfEventA = startOfEventA + widthEventA + EventAWidth
+
+    const [xPosB1, widthB1] = positionAndWidthForDates(
```
Comment (sebastianandreasson) [inline src/lib/utils.ts]: vars should be renamed to match style of xPosEventA etc, bit confusing now

File: src/lib/utils.ts [hunk 2]
```diff
@@ -62,19 +76,42 @@ const startOfEvent = (event: EventType) => event.deadlines[0].timestamp
...
+const eventsDontOverlap = (
+    eventA: EventType,
+    eventB: EventType,
+    viewportWidth: number
+) => {
+    const [xPosEventA, widthEventA] = positionAndWidthForDates(
+        eventA.deadlines.map((d) => d.timestamp),
+        viewportWidth
+    )
+    const EventAWidth =
```
Comment (sebastianandreasson) [inline src/lib/utils.ts]: lowercase for variable names

File: src/lib/utils.ts [hunk 3]
```diff
@@ -62,19 +76,42 @@ const startOfEvent = (event: EventType) => event.deadlines[0].timestamp
...
+        elementWidthOfDeadlineTitle(eventB.deadlines[0].name) >
+        elementWidthOfTitle(eventB.title)
+            ? elementWidthOfDeadlineTitle(eventB.deadlines[0].name)
+            : elementWidthOfTitle(eventB.title)
+
+    const startOfEventB = xPosB1 - EventBWidth
+    const endOfEventB = startOfEventB + widthB1 + EventBWidth
 
-    return startA.isAfter(endB) && isSameDay(startA, startB)
+    if (startOfEventA > endOfEventB || startOfEventB > endOfEventA) return true
```
Comment (sebastianandreasson) [inline src/lib/utils.ts]: here we essentially do 

```
if ( something-that-check-for-true-and-else-is-false) {
  return true
} else {
  return false
}
```

can be modified to just
```
return (startOfEventA > endOfEventB || startOfEventB > endOfEventA)
```

**Comment by sebastianandreasson** [comment]
Just these tiny fixes and then it can be merged!

File: src/lib/utils.ts (hunk 1 above)
Comment (JosefForkman) [reply src/lib/utils.ts]: I changed the names so they contain either "previous event" or "current event" in the name.

Take another look so they aren't still confusing.

Comment (sebastianandreasson) [reply src/lib/utils.ts]: this is a bit nitpicky but to explain, I didn't think A and B was the confusing part, it was that the names were inconsistent within the function. Its better now inside the function but the renaming to previous and current is not correct, they are not compared in a specific order so A can be after or before B.

**Pull request merged**